### PR TITLE
Feature king concise

### PR DESCRIPTION
**Pull request by myogh**
@mlsdpk @Awthura @aungpaing98

### CODE REVIEW

File: checkers/Board.py [hunk 1]
```diff
@@ -224,6 +312,16 @@ def move_piece(self, mouse_x, mouse_y):
...
             self.grid[selected_row][selected_col] = self.grid[
                 self.selected_piece[0]][self.selected_piece[1]]
+            
+            # update the position attribute of the piece
+            self.grid[selected_row][selected_col].position = (selected_row, selected_col)
+            
+            # make changes on the status of a piece based on its moved position
+            if self.grid[selected_row][selected_col].position[0] == 0 and player_turn == -1:
+                self.grid[selected_row][selected_col].status = 'king'
+            if self.grid[selected_row][selected_col].position[0] == 7 and player_turn == 1:
```
Comment (mlsdpk) [inline checkers/Board.py]: I think you can use only one _if_ statement here, let another one as _elif_.

File: checkers/Board.py [hunk 2]
```diff
@@ -107,8 +146,9 @@ def find_valid_moves(self, player_turn, row=None, col=None):
         # dict - {(after_captured_grid): [(capture_piece), (parent_grid)]}
         self.capture_pieces = {}
 
-        # recursively find valid moves
-        self.find_valid_moves_recursive(player_turn)
+        # recursively find valid moves of kings and men
+        player_status = self.grid[curr_row][curr_col].status
+        self.validate_grids_recursively(player_turn, player_status)
```
Comment (mlsdpk) [inline checkers/Board.py:156]: Nice, the implementation looks much cleaner than before.

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
+    "python.pythonPath": "C:\\Users\\aungm\\AppData\\Local\\Programs\\Python\\Python38\\python.exe"
+}
```
Comment (mlsdpk) [inline .vscode/settings.json]: Please add `*.vscode` to _.gitignore_ file and remove this **.vscode** folder.

**Review by mlsdpk — COMMENTED** [review]
Thanks for your pull request. I already checked the implementation and it works perfectly. I've suggested some comments and once everyone confirms, we can close this PR.

**Comment by mlsdpk** [comment]
One more question, how about this #20? If the branch _feature-king-movement_ is not needed, we can delete it.

File: checkers/Board.py (hunk 1 above)
Comment (myogh) [reply checkers/Board.py]: Agree.

File: .vscode/settings.json (hunk 3 above)
Comment (myogh) [reply .vscode/settings.json]: Will do it.

**Comment by myogh** [comment]
> One more question, how about this #20? If the branch _feature-king-movement_ is not needed, we can delete it.

Yes. We can delete it.

**Comment by myogh** [comment]
Made the corrections. @mlsdpk 
I put **.vscode** to **.gitignore**. But it's not disappearing. You could delete it here before merging to the dev.

**Comment by mlsdpk** [comment]
> Made the corrections. @mlsdpk
> I put **.vscode** to **.gitignore**. But it's not disappearing. You could delete it here before merging to the dev.

I think you need to delete the **.vscode** folder and commit it. @aungmcs

**Pull request merged**